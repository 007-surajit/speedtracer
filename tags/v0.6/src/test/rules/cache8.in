hintlet_cache_control.js

# This file tests the rule in hintlet_cache_control.js: 
#   'Consider adding a "Cache-Control: public" header to...' 
# Be careful to only trigger the specific rule you're interested in, since 
# there are several rules defined in this file.

# Input Record 1 
# Expected: HINTLET_RECORD
# Contains a Cache-Control: private  header
{"data":{"headers":{"Cache-Control":"private","Content-Length":"1461","Content-Type":"image/png","Date":"Mon, 07 Sep 1998 17:43:37 GMT","Expires":"Fri, 01 Jan 2020 00:00:00 GMT"},"resourceId":"0NetworkResourceEvent35","responseCode":200,"url":"http://www.example.com/icon.png"},"time":1,"type":23}

# Input Record 2
# Expected: none
# Contains a Cache-Control: public  header
{"data":{"headers":{"Cache-Control":"public","Content-Length":"1461","Content-Type":"image/png","Date":"Mon, 07 Sep 1998 17:43:37 GMT","Expires":"Fri, 01 Jan 2020 00:00:00 GMT"},"resourceId":"0NetworkResourceEvent35","responseCode":200,"url":"http://www.example.com/icon.png"},"time":2,"type":23}

# Input Record 3 
# Expected: none
# Contains a cookie
{"data":{"headers":{"Set-Cookie":"TESTCOOKIE","Cache-Control":"private","Content-Length":"1461","Content-Type":"image/png","Date":"Mon, 07 Sep 1998 17:43:37 GMT","Expires":"Fri, 01 Jan 2020 00:00:00 GMT"},"resourceId":"0NetworkResourceEvent35","responseCode":200,"url":"http://www.example.com/icon.png"},"time":3,"type":23}

# Input Record 4
# Expected: none
# Contains compressable resource type
{"data":{"headers":{"Cache-Control":"private","Content-Length":"1461","Content-Type":"text/css","Date":"Mon, 07 Sep 1998 17:43:37 GMT","Expires":"Fri, 01 Jan 2020 00:00:00 GMT"},"resourceId":"0NetworkResourceEvent35","responseCode":200,"url":"http://www.example.com/styles.ss"},"time":4,"type":23}