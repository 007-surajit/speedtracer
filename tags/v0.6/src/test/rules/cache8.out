{"hintletRule":"Resource Caching","timestamp":1,"description":"Consider adding a \"Cache-Control: public\" header to the following resource: http://www.example.com/icon.png","refRecord":1,"severity":3}
